src/varcache.c

```diff
@@ -258,6 +258,8 @@ void varcache_apply_startup(PktBuf *pkt, PgSocket *client)
 		struct PStr *val = get_value(&client->vars, lk);
 		if (!val)
 			continue;
+		if (strcmp(lk->name, "in_hot_standby") == 0 || strcmp(lk->name, "default_transaction_read_only") == 0)
```

Comment:
This feature was first drafted before the implementation of replication connections. I'm unsure if this adjustment to satisfy test_replication.py is appropriate here, complete, and without edge-cases.